src/clj/clojurians_log/views.clj

```diff
@@ -21,11 +21,32 @@
    ;; styling over in clean Garden or Garden+Tachyons.
    [:link {:href "/css/legacy.css", :rel "stylesheet", :type "text/css"}]])
 
-(defn- log-page-header [{:data/keys [channel]}]
+(defn channel-day-offset
+  "Given a list of [date msg-count] pairs, return `date` of the entry that is
+  `offset` positions away. Returns nil if the applying the offset goes out of
+  bounds."
+  [channel-days today offset]
+
+  (as-> channel-days $
+    (some (fn [[index [a-date msg-count]]] (when (and (= a-date today)
+                                                     (not (zero? msg-count)))
+                                            index))
+          (map vector (range) $))
+    (+ $ offset)
+    (nth channel-days $ nil)
+    (first $)))
```

Comment:
Since you're threading already, how about splitting that `(some ,,, (map ,,,))` into separate steps?

```
(as-> channel-days $
  (map vector (range) $)
  (some ,,,)
  (+ ,,,)
)
```

Reply:
Good point!